recipes/fsx_mount.rb

```diff
@@ -0,0 +1,81 @@
+#
+# Cookbook Name:: aws-parallelcluster
+# Recipe:: fsx_mount
+#
+# Copyright 2013-2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+fsx_shared_dir = node['cfncluster']['cfn_fsx_shared_dir'].split(',')[0]
+fsx_fs_id = node['cfncluster']['cfn_fsx']
+
+lustre_kmod_rpm = "#{node['cfncluster']['sources_dir']}/kmod-lustre-client-#{node['cfncluster']['lustre']['version']}.x86_64.rpm"
+lustre_client_rpm = "#{node['cfncluster']['sources_dir']}/lustre-client-#{node['cfncluster']['lustre']['version']}.x86_64.rpm"
+
+# Check to see if FSx is created
+if fsx_shared_dir != "NONE"
+
+  # Path needs to be fully qualified, for example "shared/temp" becomes "/shared/temp"
+  if !fsx_shared_dir.start_with?("/")
+    fsx_shared_dir = "/" + fsx_shared_dir
+  end
+
+  # Create the shared directories
+  directory fsx_shared_dir do
+    owner 'root'
+    group 'root'
+    mode '1777'
+    action :create
+  end
+
+  # Get Lustre Kernel Module RPM
```

Comment:
download and installation should be done early in the process, during packer step

Reply:
I agree, we should avoid dependency from external website as far as possible, to simplify an eventual Proxy configuration.

Reply:
Move download and installation in base_install.rb

Reply:
done

Reply:
done